coreAPI/src/main/java/net/java/games/input/PomProperties.java

```diff
@@ -0,0 +1,128 @@
+package net.java.games.input;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.logging.Logger;
+
+/**
+ * <p>Provides access to the Maven properties of this implementation.</p>
+ *
+ * <p>
+ *     These properties are loaded from the {@code pom.properties} file, which is generated by Maven during the build
+ *     process.
+ * </p>
+ *
+ * <p>
+ *     Click <a href="https://maven.apache.org/guides/introduction/introduction-to-the-pom.html#Project_Properties">here</a>
+ *     for more information on the Maven properties file.
+ * </p>
+ */
+public final class PomProperties {
+    /** Path to the Maven properties file, within the JAR. */
+    private final static String PROPERTIES_FILE_PATH = "/META-INF/maven/net.java.jinput/coreapi/pom.properties";
+
+    /** Singleton instance. */
+    private static final PomProperties INSTANCE = new PomProperties();
+
+    /** {@link Properties} object containing either the default or loaded properties. */
+    private final Properties properties = new Properties();
+
+
+    /** Private constructor to prevent instantiation. */
+    private PomProperties() {
+        properties.setProperty("artifactId", "Unknown");
+        properties.setProperty("groupId", "Unknown");
+        properties.setProperty("version", "Unversioned");
+
+        loadProperties();
+    }
+
+    /** Attempts to load the Maven properties file into {@code properties}. */
+    private void loadProperties() {
+        try (
+            final InputStream is = PomProperties.class.getResourceAsStream(PROPERTIES_FILE_PATH);
+        ) {
+            Objects.requireNonNull(is);
+            properties.load(is);
+        } catch (final IOException | NullPointerException e) {
+            final Logger logger = Logger.getLogger(PomProperties.class.getName());
+            logger.warning("Failed to load properties file: " + PROPERTIES_FILE_PATH);
```

Comment:
This should never occur, as far as I'm aware. However, I thought it better to warn developers about the failure rather than silently ignoring/handling it as the original implementation did.